selfdrive/locationd/locationd.cc

```diff
@@ -239,36 +245,33 @@ void Localizer::handle_gps(double current_time, const cereal::GpsLocationData::R
 }
 
 void Localizer::handle_car_state(double current_time, const cereal::CarState::Reader& log) {
-  this->speed_counter++;
-
-  if (this->speed_counter % SENSOR_DECIMATION == 0) {
-    this->kf->predict_and_observe(current_time, OBSERVATION_ODOMETRIC_SPEED, { (VectorXd(1) << log.getVEgoRaw()).finished() });
-    this->car_speed = std::abs(log.getVEgo());
-    if (this->car_speed < 1e-3) {
-      this->kf->predict_and_observe(current_time, OBSERVATION_NO_ROT, { Vector3d(0.0, 0.0, 0.0) });
-    }
+  //this->kf->predict_and_observe(current_time, OBSERVATION_ODOMETRIC_SPEED, { (VectorXd(1) << log.getVEgoRaw()).finished() });
+  this->car_speed = std::abs(log.getVEgo());
```

Comment:
Why not vEgoRaw?

Reply:
It's not put into the filter so I thought smoothed is better, but can change if you want.

Reply:
The filtered version is a little weird around sudden stops where it can overshoot and become a little bit negative. In this case I would recommend the raw speed.

If you only care about standstill I would use the dedicated standstill flag. For some cars this is a different signal and is a more accurate at very low speeds (where the wheel speeds would be zero even though you’re moving)